omod/src/test/java/org/openmrs/module/idgen/rest/controller/AutoGenerationOptionControllerTest.java

```diff
@@ -0,0 +1,145 @@
+			allAutoGenerationOptions += identifierSourceService.getAutoGenerationOptions(patientIdentifierType).size();
+		}
+		return allAutoGenerationOptions;
+	}
+	
+	@Test
+	public void shouldGetAnAutoGenerationOptionByPatientIdentifierType() throws Exception {
+		MockHttpServletRequest req = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		SimpleObject result = deserialize(handle(req));
+		
+		AutoGenerationOption autoGenerationOption = identifierSourceService
+		        .getAutoGenerationOption(Context.getPatientService().getPatientIdentifierTypeByUuid(getUuid()));
+		
+		assertEquals(autoGenerationOption.getIdentifierType().getUuid(),
+		    PropertyUtils.getProperty(result, "identifierType"));
+		assertEquals(autoGenerationOption.getLocation().getUuid(), PropertyUtils.getProperty(result, "location"));
+		assertEquals(autoGenerationOption.getSource().getUuid(), PropertyUtils.getProperty(result, "source"));
+	}
+	
+	@Test
+	public void shouldCreateNewAutoGenerationOption() throws Exception {
+		long originalCount = getAllCount();
+		boolean automaticGenerationEnabled = true;
+		boolean manualEntryEnabled = true;
+		
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("identifierType", getUuid());
+		autoGenerationOption.add("location", "8d6c993e-c2cc-11de-8d13-0010c6dffd0f");
+		autoGenerationOption.add("source", "0d47284f-9e9b-4a81-a88b-8bb42bc0a901");
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject newAutoGenerationOption = deserialize(handle(req));
+		
+		assertNotNull(PropertyUtils.getProperty(newAutoGenerationOption, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+	}
+	
+	@Test
+	public void shouldEditAnAutoGenerationOption() throws Exception {
+		final String newLocation = "9356400c-a5a2-4532-8f2b-2361b3446eb8";
+		final String newSource = "0d47284f-9e9b-4a81-a88b-8bb42bc0a902";
+		boolean automaticGenerationEnabled = false;
+		boolean manualEntryEnabled = false;
+		SimpleObject autoGenerationOption = new SimpleObject();
+		autoGenerationOption.add("location", newLocation);
+		autoGenerationOption.add("source", newSource);
+		autoGenerationOption.add("manualEntryEnabled", manualEntryEnabled);
+		autoGenerationOption.add("automaticGenerationEnabled", automaticGenerationEnabled);
+		
+		String json = new ObjectMapper().writeValueAsString(autoGenerationOption);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+		handle(req);
+		assertEquals(newLocation,
+		    identifierSourceService
+		            .getAutoGenerationOption(Context.getPatientService().getPatientIdentifierTypeByUuid(getUuid()))
```

Comment:
Declare `Context.getPatientService().getPatientIdentifierTypeByUuid(getUuid())` at the top.